lightning/src/ln/features.rs

```diff
@@ -926,6 +970,55 @@ mod tests {
 		assert!(features.supports_payment_secret());
 	}
 
+	#[test]
+	fn set_custom_bits() {
+		let mut features = InvoiceFeatures::empty();
+		features.set_variable_length_onion_optional();
+		assert_eq!(features.flags[1], 0b00000010);
+
+		assert!(features.set_custom_bit(255).is_err());
+		assert!(features.set_custom_bit(256).is_ok());
+		assert!(features.set_custom_bit(258).is_ok());
+		assert_eq!(features.flags[31], 0b00000000);
+		assert_eq!(features.flags[32], 0b00000101);
+
+		let known_bit = <sealed::InvoiceContext as sealed::PaymentSecret>::EVEN_BIT;
+		let byte_offset = <sealed::InvoiceContext as sealed::PaymentSecret>::BYTE_OFFSET;
+		assert_eq!(byte_offset, 1);
+		assert_eq!(features.flags[byte_offset], 0b00000010);
+		assert!(features.set_custom_bit(known_bit).is_err());
+		assert_eq!(features.flags[byte_offset], 0b00000010);
```

Comment:
Does this test anything new since `<sealed::InvoiceContext as sealed::PaymentSecret>::EVEN_BIT` = 14 which gets caught by the initial <256 check?